docs/administration/back_office/search_suggestion.md

```diff
@@ -0,0 +1,151 @@
+During this event, you can add, remove, or replace suggestions by updating its `SuggestionCollection`.
+After this event, the suggestion collection is sorted by score and truncated to [`result_limit`](#configuration) items.
+
+!!! tip
+
+    You can list listeners and subscribers with the following command:
+    ``` shell
+    php bin/console debug:event BuildSuggestionCollectionEvent
+    ```
+
+The following example is boosting Product suggestions.
+It's a subscriber passing after the default one (thanks to a priority below zero), adding matching products at a score above the earlier Content suggestions, and avoiding duplicates.
+
+- If this suggestion source find a count of `result_limit` matching products or more, there is only those products in the suggestion.
+- If it finds less than `result_limit` products, those products are on top of the suggestion followed by other suggestion source items to reach the limit.
+- If it doesn't find any matching product, only default source suggestion is shown.
+
+This example event subscriber is implemented in the file `src/EventSubscriber/MySuggestionEventSubscriber.php`, uses [`ProductService::findProducts`](product_api.md#products), and return the received event after having manipulated it `SuggestionCollection`:
+``` php
+[[= include_file('code_samples/back_office/search/src/EventSubscriber/MySuggestionEventSubscriber.php') =]]
+```
+
+To have the logger injected thanks to the `LoggerAwareTrait`, this subscriber must be registered as a service:
+``` yaml
+services:
+    #…
+[[= include_file('code_samples/back_office/search/config/append_to_services.yaml', 2, 3) =]]
+```
+
+To represent the product suggestion data, a `ProductSuggestion` class is created in `src/Search/Model/Suggestion/ProductSuggestion.php`:
+``` php
+[[= include_file('code_samples/back_office/search/src/Search/Model/Suggestion/ProductSuggestion.php') =]]
+```
+
+This representation need a normalizer to be transformed into JSON.
+`ProductSuggestionNormalizer::supportsNormalization` associates with `ProductSuggestion`.
+`ProductSuggestionNormalizer::normalize` return an array of scalar values which can be transformed into a JSON object.
+Alongside data about the product, this array must have a `type` key which value is an identifier used later for rendering.
+In `src/Search/Serializer/Normalizer/Suggestion/ProductSuggestionNormalizer.php`:
+
+``` php
+[[= include_file('code_samples/back_office/search/src/Search/Serializer/Normalizer/Suggestion/ProductSuggestionNormalizer.php') =]]
+```
+
+This normalizer is added to suggestion normalizers by decorating `ibexa.search.suggestion.serializer` and redefining its list of normalizers:
+
+``` yaml
+services:
+    #…
+[[= include_file('code_samples/back_office/search/config/append_to_services.yaml', 4, 20) =]]
+```
+
+!!! tip
+
+    At this point, it's possible to test the suggestion JSON.
+    The route is `/suggestion` with a `query` GET parameter worth the searched text.
```

Comment:
```suggestion
    The route is `/suggestion` with a `query` GET parameter worth the searched text.
```

worth? or with?

Reply:
@dabrt "worth", like "has the value".

```suggestion
    The route is `/suggestion` with a `query` GET parameter to pass the searched text.
```
```suggestion
    The route is `/suggestion` with a `query` GET parameter for the searched text.
```

Reply:
Done by using "for" in 984825cd6a549c9f03f864ec533ee605f6818e54